src/compass/s1_geocode_topo.py

```diff
@@ -0,0 +1,144 @@
+
+    '''
+    module_name = get_module_name(__file__)
+    info_channel = journal.info(f"{module_name}.run")
+    info_channel.log(f"Starting {module_name} burst")
+
+    # Start tracking processing time
+    t_start = time.time()
+
+    # common initializations
+    dem_raster = isce3.io.Raster(cfg.dem)
+    epsg = dem_raster.get_epsg()
+    proj = isce3.core.make_projection(epsg)
+    ellipsoid = proj.ellipsoid
+    image_grid_doppler = isce3.core.LUT2d()
+    threshold = cfg.geo2rdr_params.threshold
+    iters = cfg.geo2rdr_params.numiter
+    blocksize = cfg.geo2rdr_params.lines_per_block
+    output_epsg = cfg.geocoding_params.output_epsg
+    output_format = cfg.geocoding_params.output_format
+
+    # process one burst only
+    burst = cfg.bursts[0]
+    date_str = burst.sensing_start.strftime("%Y%m%d")
+    burst_id = burst.burst_id
+    geo_grid = cfg.geogrids[burst_id]
+
+    os.makedirs(cfg.output_dir, exist_ok=True)
+
+    scratch_path = f'{cfg.scratch_path}/{burst_id}/{date_str}'
+    os.makedirs(scratch_path, exist_ok=True)
+
+    radar_grid = burst.as_isce3_radargrid()
+    orbit = burst.orbit
+
+    # Initialize input/output path
+    input_path = f'{cfg.product_path}/{burst_id}/{date_str}'
+    output_path = input_path
+    if fetch_from_scratch:
+        input_path = f'{cfg.scratch_path}/{burst_id}/{date_str}'
+    os.makedirs(output_path, exist_ok=True)
+
+    # Initialize geocode object
+    geo = isce3.geocode.GeocodeFloat32()
+    geo.orbit = orbit
+    geo.ellipsoid = ellipsoid
+    geo.doppler = image_grid_doppler
+    geo.threshold_geo2rdr = threshold
+    geo.numiter_geo2rdr = iters
+    geo.lines_per_block = blocksize
+    geo.geogrid(geo_grid.start_x, geo_grid.start_y,
+                geo_grid.spacing_x, geo_grid.spacing_y,
+                geo_grid.width, geo_grid.length, geo_grid.epsg)
+
+    # Get the topo layers to compute
+    topo_layers = {'x': cfg.rdr2geo_params.compute_longitude,
+                   'y': cfg.rdr2geo_params.compute_latitude,
+                   'z': cfg.rdr2geo_params.compute_height,
+                   'incidence': cfg.rdr2geo_params.compute_incidence_angle,
+                   'localIncidence': cfg.rdr2geo_params.compute_local_incidence_angle,
+                   'localPsi': cfg.rdr2geo_params.compute_azimuth_angle}
+
+    input_rasters = [
+        isce3.io.Raster(f'{input_path}/{fname}.rdr')
+        if enabled else None
+        for fname, enabled in topo_layers.items()]
+    output_rasters = [
+        isce3.io.Raster(f'{output_path}/{fname}.geo',
+                        geo_grid.width, geo_grid.length, 1, gdal.GDT_Float32,
+                        output_format)
+        if enabled else None
+        for fname, enabled in topo_layers.items()
+    ]
+    # Filter None from list
+    input_rasters = [raster for raster in input_rasters if raster is not None]
+    output_rasters = [raster for raster in output_rasters if raster is not None]
```

Comment:
```suggestion
    input_rasters = [
        isce3.io.Raster(f'{input_path}/{fname}.rdr')
        for fname, enabled in topo_layers.items() if enabled]        
    output_rasters = [
        isce3.io.Raster(f'{output_path}/{fname}.geo',
                        geo_grid.width, geo_grid.length, 1, gdal.GDT_Float32,
                        output_format)
        for fname, enabled in topo_layers.items() if enabled]
```
No need to remove `None` by moving `enabled` check after `for` loop.